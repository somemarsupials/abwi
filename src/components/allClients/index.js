export { default as ClientsPage } from './clientsList';
export { default as ClientsList } from './clientsPage';
