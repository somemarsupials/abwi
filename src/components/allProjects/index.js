export { default as List } from './list';
export { default as Page } from './page';
