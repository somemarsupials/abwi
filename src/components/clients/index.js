export { default as ClientPage } from './clientPage';
export { default as ClientDetail } from './clientDetail';
