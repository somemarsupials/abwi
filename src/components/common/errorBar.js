import React from 'react';

export default function(props) {
  return (
    <div className="alert alert-danger" role="alert">
      <p>
        <span className="font-weight-bold">{props.text}</span>
        { props.detail ? <span>&emsp;|&emsp;{props.detail}</span> : null }
      </p>
    </div>
  );
};
