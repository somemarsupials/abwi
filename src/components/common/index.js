export { default as TitleBar } from './titleBar';
export { default as ErrorBar } from './errorBar';
