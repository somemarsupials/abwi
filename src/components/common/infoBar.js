import React from 'react';

export default function(props) {
  return (
    <div className="alert alert-info">
      <p>{props.text}</p>
    </div>
  );
};

