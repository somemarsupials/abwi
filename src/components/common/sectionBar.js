import React from 'react';

export default function(props) {
  return (
    <div className="page-header">
      <h3>{props.text}</h3>
    </div>
  );
};
