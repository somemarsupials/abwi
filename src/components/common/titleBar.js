import React from 'react';

export default function(props) {
  return (<h1>{props.text}</h1>);
};
