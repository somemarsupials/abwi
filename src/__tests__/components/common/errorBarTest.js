import React from 'react';
import { shallow } from 'enzyme';

import { ErrorBar } from '../../../components/common';

describe('ErrorBar', () => {
  let errorBar;

  describe('when passed detailed information', () => {
    beforeEach(() => {
      errorBar = shallow(
        <ErrorBar message={'Nerdy error!'} detail={'Details...'} />
      );
    });

    it('correctly renders', () => {
      expect(errorBar).toMatchSnapshot();
    });
  });

  describe('when passed no detail', () => {
    beforeEach(() => {
      errorBar = shallow(
        <ErrorBar message={'Error!'} />);
    });

    it('correctly renders', () => {
      expect(errorBar).toMatchSnapshot();
    });
  });
});

