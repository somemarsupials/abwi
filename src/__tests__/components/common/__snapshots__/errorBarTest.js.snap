// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBar when passed detailed information correctly renders 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <p>
    <span
      className="font-weight-bold"
    >
      Nerdy error!
    </span>
    <span>
       | 
      Details...
    </span>
  </p>
</div>
`;

exports[`ErrorBar when passed no detail correctly renders 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <p>
    <span
      className="font-weight-bold"
    >
      Error!
    </span>
  </p>
</div>
`;
